### PR TITLE
use gnu cp to take/restore snapshot

### PR DESCRIPTION
**Pull request by dkeven**
Signed-off-by: dkven <kevendeng@yunify.com>

**What type of PR is this?**
/kind improvement

**What this PR does / why we need it**:
use GNU cp to speed up snapshotting/resotring

**Does this PR introduce a user-facing change?**:

NONE

### CODE REVIEW

File: pkg/hostpath/hostpath.go [hunk 1]
```diff
@@ -476,7 +477,8 @@ func (hp *hostPath) loadFromSnapshot(size int64, snapshotId, destPath string, mo
 	var cmd []string
 	switch mode {
 	case mountAccess:
-		cmd = []string{"tar", "zxvf", snapshotPath, "-C", destPath}
+		// destPath has already been created at this time
+		cmd = []string{"/bin/sh", "-c", fmt.Sprintf("cp -r %s/* %s", snapshotPath, destPath)}
```
Comment (stoneshi-yunify) [inline pkg/hostpath/hostpath.go]: Why not use `[]string{"cp", "-r", snapshotPath, destPath}`?

Comment (dkeven) [reply pkg/hostpath/hostpath.go]: Before copying, the `destPath` is already created by

https://github.com/stoneshi-yunify/csi-driver-host-path/blob/311b0e5eb4302391cd04dfcadcc5404163b326a6/pkg/hostpath/controllerserver.go#L139

Therefore, with `[]string{"cp", "-r", snapshotPath, destPath}`, the `snapshotPath`(xxxx.snap) will be copied as a whole, nested under `destPath`, but what we actually need are the contents of `snapshotPath`, so we need to explicitly set the contents to be copied.

Comment (stoneshi-yunify) [reply pkg/hostpath/hostpath.go]: I see. However, the string `%s/*` can not match hidden file, keep in mind that we're developing a common usage hostpath csi, not just for ksv.

if the dest dir exists, try use the cp's `-T` option. ref: https://stackoverflow.com/questions/23698183/how-to-force-cp-to-overwrite-directory-instead-of-creating-another-one-inside

Comment (dkeven) [reply pkg/hostpath/hostpath.go]: Got it, I'll try that, thanks.

**Comment by dkeven** [comment]
I've tested the new approach, hidden file/file attributes/file ownership, etc can all be snapshotted/restored correctly now.
/ok-to-test

File: pkg/hostpath/hostpath.go [hunk 2]
```diff
@@ -476,7 +477,10 @@ func (hp *hostPath) loadFromSnapshot(size int64, snapshotId, destPath string, mo
 	var cmd []string
 	switch mode {
 	case mountAccess:
-		cmd = []string{"tar", "zxvf", snapshotPath, "-C", destPath}
+		// destPath has already been created at this time
+		// so use "-T" to copy contents under snapshootPath rather than the dir itself
+		// also, we use "-a" to archive everything(ownership/mode/xattr/link...) of files
```
Comment (stoneshi-yunify) [inline pkg/hostpath/hostpath.go]: this two lines of comments are unnecessary

Comment (dkeven) [reply pkg/hostpath/hostpath.go]: OK, I’ve deleted them.

**Review by stoneshi-yunify — APPROVED** [review]

**Pull request merged**